state/pruner.go

```diff
@@ -92,45 +111,34 @@ func (p *Pruner) SetObserver(obs PrunerObserver) {
 
 func (p *Pruner) OnStart() error {
 	go p.pruneBlocksRoutine()
-	go p.pruneABCIResRoutine()
+	// We only care about pruning ABCI results if the data companion has been
+	// enabled.
+	if p.dcEnabled {
+		go p.pruneABCIResRoutine()
```

Comment:
Side note: I don't like the "routine" naming style.
It's, like, a programming language, every function is a routine of some sort.